api-inference-community/manage.py

```diff
@@ -39,6 +42,84 @@ def create_docker(name: str) -> str:
     return tag
 
 
+def resolve_dataset(args, task: str):
+    import datasets
+
+    builder = datasets.load_dataset_builder(
+        args.dataset_name, use_auth_token=args.token
+    )
+
+    if args.dataset_config is None:
+        args.dataset_config = builder.config_id
+        print(f"Inferred dataset_config {args.dataset_config}")
+
+    splits = builder.info.splits
+    if splits is not None:
+        if args.dataset_split not in splits:
+            raise ValueError(
+                f"The split `{args.dataset_split}` is not a valid split, please choose from {','.join(splits.keys())}"
+            )
+
+    task_templates = builder.info.task_templates
+    if task_templates is not None:
+        for task_template in task_templates:
+            if task_template.task == task:
+                args.dataset_column = task_template.audio_file_path_column
+                print(f"Inferred dataset_column {args.dataset_column}")
+    return (
+        args.dataset_name,
+        args.dataset_config,
+        args.dataset_split,
+        args.dataset_column,
+    )
```

Comment:
No strong opinion, but consider using dataclasses instead of tuples of 4 elements. That way we can have better typings and make the program more readable in my opinion

Reply:
Ok, I'll keep that in mind.

Definitely getting borderline here. I made the current choice because adding types also has drawbacks (more complex to read and understand).

I think the biggest issue I have is with `_column` which for some pipelines won't be unique.